helix-term/src/application.rs

```diff
@@ -1014,6 +1014,17 @@ impl Application {
                             .collect();
                         Ok(json!(result))
                     }
+                    Ok(MethodCall::RegisterCapability(_params)) => {
+                        log::warn!("Ignoring a client/registerCapability request because dynamic capability registration is not enabled");
```

Comment:
I do like how the neovim error says to report the issue upstream: https://github.com/neovim/nvim-lspconfig/pull/471/files#diff-0e1bd346624fa540fa153e867fbc2036ced5c83d5cd7d79d9308df1328b7c879R58-R59

That way if someone spots it, it does actually get fixed upstream rather than considered a helix bug